ebos/ecltransmissibility.hh

```diff
@@ -475,14 +475,11 @@ private:
         if (insideFaceIdx > 3) { // top or or bottom
             Scalar mult = 1e20;
             unsigned cartElemIdx = insideCartElemIdx;
+            assert(insideFaceIdx==5); // as insideCartElemIdx < outsideCartElemIdx holds
             // pick the smallest multiplier while looking down the pillar untill reaching the other end of the connection
             // for the inbetween cells we apply it from both sides
             while (cartElemIdx != outsideCartElemIdx) {
-                if (insideFaceIdx == 4 || cartElemIdx !=insideCartElemIdx )
-                    mult = std::min(mult, transMult.getMultiplier(cartElemIdx, Opm::FaceDir::ZMinus));
-                if (insideFaceIdx == 5 || cartElemIdx !=insideCartElemIdx)
-                    mult = std::min(mult, transMult.getMultiplier(cartElemIdx, Opm::FaceDir::ZPlus));
-
+                mult = std::min(mult, transMult.getMultiplier(cartElemIdx, Opm::FaceDir::ZPlus));
```

Comment:
This looks dangerous.  Would you not be missing `MULTZ-` this way?

Reply:
Could be (I am able to judge this), but then that was the case also before.

I think the original code gave the impression that each face might be processed for both sides. But that has never been the case. We always processed Z faces only from low to high cartesian indices here, hence ZPLUS.

Reply:
Indeed, looking at [TransMult.cpp#L91-L97](/OPM/opm-common/blob/master/src/opm/parser/eclipse/EclipseState/Grid/TransMult.cpp#L91-L97), I totally agree that this is very dangerous as the multiplier is only stored for one direction and we might miss it. Maybe @joakim-hove can shed some light which direction is chosen and when.

Reply:
I correct myself: there might be mutlipiers stored for each direction. Hence you might have found a bug that has been lingering for a long time.

Reply:
I don't really know anything about this code - I just accidentally stumbled over a problem here - which I am not familiar with at all.